pydicom/util/dump.py

```diff
@@ -1,72 +1,106 @@
-# Copyright 2008-2018 pydicom authors. See LICENSE file for details.
+# Copyright 2008-2021 pydicom authors. See LICENSE file for details.
 """Utility functions used in debugging writing and reading"""
 
 from io import BytesIO
+import pathlib
+import sys
+from typing import Union, Optional, BinaryIO, TYPE_CHECKING
 
+if TYPE_CHECKING:  # pragma: no cover
+    from pydicom.dataset import Dataset
 
-def print_character(ordchr):
+
+def print_character(ordchr: int) -> str:
     """Return a printable character, or '.' for non-printable ones."""
     if 31 < ordchr < 126 and ordchr != 92:
         return chr(ordchr)
-    else:
-        return '.'
+
+    return "."
 
 
-def filedump(filename, start_address=0, stop_address=None):
-    """Dump out the contents of a file to a
-       standard hex dump 16 bytes wide"""
+def filedump(
+    filename: Union[str, pathlib.Path],
+    start_address: int = 0,
+    stop_address: Optional[int] = None,
+) -> str:
+    """Dump out the contents of a file to a standard hex dump 16 bytes wide"""
 
-    fp = open(filename, 'rb')
-    return hexdump(fp, start_address, stop_address)
+    with open(filename, "rb") as f:
+        return hexdump(f, start_address, stop_address)
 
 
-def datadump(data):
-    stop_address = len(data) + 1
-    fp = BytesIO(data)
-    print(hexdump(fp, 0, stop_address))
```

Comment:
This one should just print the results from `bytes` input. I figured the functions were intended to handle different input types so I changed it to return the `str` rather than printing. Really it would make more sense to just have a single function that can handle the various input types.